File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/GetTCFDataBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton
import com.usercentrics.sdk.flutter.serializer.serialize

internal class GetTCFDataBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "getTCFData"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        usercentrics.instance.getTCFData {
            result.success(it.serialize())
        }
    }
}

File: ios/usercentrics_sdk/Sources/usercentrics_sdk/Bridge/GetTCFDataBridge.swift
import Usercentrics
import Flutter

struct GetTCFDataBridge : MethodBridge {

    let name: String = "getTCFData"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        usercentrics.shared.getTCFData() {
            result($0.serialize())
        }
    }
}
